### ISSUE TITLE
It's not able to select all day box when a selectConstraint is specified

### ISSUE THREAD

**Issue by jfelixalba** [fullcalendar/fullcalendar#6039]
### Reduced Test Case

https://codesandbox.io/s/condescending-microservice-xoy5j

### Bug Description

It's not able to select all day box when a selectConstraint is specified, I have tried different ways, specifying the word 'businessHours' or specifying the same business hours with the flag allDay = true.

Describe how to recreate the bug.
To reproduce the bug you just need to create a calendar component with a specific business hour and with a constraint, select week view, and try to select all day and it is enable visibly, but disabled to select.

What do you expect to happen?
You can select time slots that are in the business hours or all day slot.

What happens instead?
You cannot select all day slot

**Comment by acerix** [comment]
It doesn't seem like a bug since an allDay event does not fit in the business hours, but could be useful as a new option.  You could use `eventAllow` to do this instead of a constraint.

**Comment by jfelixalba** [comment]
But selectAllow function does not return if it is a business hour or non-business hour. Just start, end and all day fields are visible in the callback.

**Comment by acerix** [comment]
Yes, you'd need to manually compare the selection times and day of week to the business hours setting.

**Comment by jfelixalba** [comment]
wouldn't it be better if more details are returned, or there's a reason to not return more info about the event/time slot info?

**Comment by gragia79** [comment]
I'm using version 5.
I have the same problem.
should it work now?
my use in test:

```
        eventConstraint: 'businessHours',
        businessHours: [
             {
                 daysOfWeek: [ 4, 5 ],
                 start: '08:00', 
                 end: '12:00',
                 dow: [2]
             },
             {
                 daysOfWeek: [ 3, 5 ],
                 start: '13:00', 
                 end: '17:00', 
                 dow: [4]
             }
         ],
```
My goal is to hide days of the week. example: Show = Monday, Wednesday Hide = Tuesday, Thursday, Friday
This is possible?
